sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_models.py

```diff
@@ -39,29 +42,259 @@ class ConfigurationSetting(Model):
-            content_type=key_value.content_type,
             value=key_value.value,
+            content_type=key_value.content_type,
             last_modified=key_value.last_modified,
             tags=key_value.tags,
             read_only=key_value.locked,
             etag=key_value.etag,
         )
+
+    def _to_generated(self):
+        # type: (...) -> KeyValue
+        return KeyValue(
+            key=self.key,
+            label=self.label,
+            value=self.value,
+            content_type=self.content_type,
+            last_modified=self.last_modified,
+            tags=self.tags,
+            locked=self.read_only,
+            etag=self.etag,
+        )
+
+
+class FeatureFlagConfigurationSetting(
+    ConfigurationSetting
+):  # pylint: disable=too-many-instance-attributes
+    """A feature flag configuration value.
+    Variables are only populated by the server, and will be ignored when
+    sending a request.
+    :ivar etag: Entity tag (etag) of the object
+    :vartype etag: str
+    :ivar key:
+    :vartype key: str
+    :ivar enabled:
+    :vartype enabled: bool
+    :param filters:
+    :type filters: list[dict[str, Any]]
+    :param label:
+    :type label: str
+    :param content_type:
+    :type content_type: str
+    :ivar last_modified:
+    :vartype last_modified: datetime
+    :ivar read_only:
+    :vartype read_only: bool
+    :param tags:
+    :type tags: dict[str, str]
+    """
+
+    _attribute_map = {
+        "etag": {"key": "etag", "type": "str"},
+        "key": {"key": "key", "type": "str"},
+        "label": {"key": "label", "type": "str"},
+        "content_type": {"key": "content_type", "type": "str"},
+        "value": {"key": "value", "type": "str"},
+        "last_modified": {"key": "last_modified", "type": "iso-8601"},
+        "read_only": {"key": "read_only", "type": "bool"},
+        "tags": {"key": "tags", "type": "{str}"},
+    }
+    key_prefix = ".appconfig.featureflag/"
+    _feature_flag_content_type = (
+        "application/vnd.microsoft.appconfig.ff+json;charset=utf-8"
+    )
+    kind = "FeatureFlag"
+
+    def __init__(self, key, enabled, filters=None, **kwargs):
+        # type: (str, bool, Optional[List[Dict[str, Any]]]) -> None
+        super(FeatureFlagConfigurationSetting, self).__init__(**kwargs)
+        self.key = key
+        if not key.startswith(self.key_prefix):
+            self.key = self.key_prefix + key
+        self.enabled = enabled
+        self.label = kwargs.get("label", None)
+        self.content_type = kwargs.get("content_type", self._feature_flag_content_type)
+        self.last_modified = kwargs.get("last_modified", None)
+        self.tags = kwargs.get("tags", {})
+        self.read_only = kwargs.get("read_only", None)
+        self.etag = kwargs.get("etag", None)
+        self.description = kwargs.get("description", None)
+        self.display_name = kwargs.get("display_name", None)
+        self.filters = filters or []
```

Comment:
How do you handle use modifies filters property and value property needs to get updated as well?

Reply:
The `value` parameter is recreated by the `_to_generated` method.

Reply:
_to_generated() is called when you talk to server.

What if:

```py
config_setting = FeatureFlagConfigurationSetting()     # with some value
config_setting.filters = <new value>
print(config_setting.value)       <-- expect updated value
```

Reply:
This will print `None`, a FeatureFlagConfigurationSetting value is stored in the `enabled` variable.

Reply:
Taken offline.